features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/impl/DatadogTracerAdapterTest.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+package com.datadog.android.trace.impl
+
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.trace.api.scope.DataScopeListener
+import com.datadog.android.trace.api.span.DatadogSpanBuilder
+import com.datadog.android.trace.impl.internal.DatadogPropagationAdapter
+import com.datadog.android.trace.impl.internal.DatadogScopeAdapter
+import com.datadog.android.trace.impl.internal.DatadogScopeListenerAdapter
+import com.datadog.android.trace.impl.internal.DatadogSpanAdapter
+import com.datadog.android.trace.impl.internal.DatadogTracerAdapter
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.trace.bootstrap.instrumentation.api.AgentScope
+import com.datadog.trace.bootstrap.instrumentation.api.AgentSpan
+import com.datadog.trace.bootstrap.instrumentation.api.AgentTracer
+import com.datadog.trace.bootstrap.instrumentation.api.ScopeSource
+import fr.xgouchet.elmyr.annotation.BoolForgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.any
+import org.mockito.kotlin.isA
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class DatadogTracerAdapterTest {
+
+    private lateinit var testedTracer: DatadogTracerAdapter
+
+    @Mock
+    lateinit var mockSdk: FeatureSdkCore
+
+    @Mock
+    lateinit var mockTracer: AgentTracer.TracerAPI
+
+    @Mock
+    lateinit var mockSpan: AgentSpan
+
+    @Mock
+    lateinit var mockDatadogSpan: DatadogSpanAdapter
+
+    @Mock
+    lateinit var mockScope: AgentScope
+
+    @Mock
+    lateinit var mockSpanBuilder: AgentTracer.SpanBuilder
+
+    @StringForgery
+    lateinit var fakeString: String
+
+    @BoolForgery
+    private var fakeBool: Boolean = false
+
+    @BeforeEach
+    @Suppress("DEPRECATION")
```

Comment:
what is deprecated here?

Reply:
AgentTracer's `buildSpan(String)` method. Moved closer to class site